src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -656,6 +656,70 @@ case object LOG3 extends LogOp(0xa3)
 case object LOG4 extends LogOp(0xa4)
 
 
+case object CREATE extends OpCode(0xf0, 3, 1, G_create) {
+  protected def exec[W <: WorldStateProxy[W, S], S <: Storage[S]](state: ProgramState[W, S]): ProgramState[W, S] = {
+    val (Seq(endowment, inOffset, inSize), stack1) = state.stack.pop(3)
+
+    val validCall = state.env.callDepth < OpCode.MaxCallDepth && endowment <= state.ownBalance
+
+    if (!validCall) {
+      val stack2 = stack1.push(UInt256.Zero)
+      state.withStack(stack2).step()
+    } else {
+
+      val (initCode, memory1) = state.memory.load(inOffset, inSize)
+      val (newAddress, world1) = state.world.newAddress(state.env.ownerAddr)
+      val world2 = world1.transfer(state.env.ownerAddr, newAddress, endowment)
+
+      val newEnv = ExecEnv(
+        newAddress,
+        state.env.callerAddr,
+        state.env.originAddr,
+        state.env.gasPrice,
+        ByteString.empty,
+        endowment,
+        Program(initCode),
+        state.env.blockHeader,
+        state.env.callDepth + 1
+      )
+
+      //to avoid calculating this twice, we could adjust state.gas prior to execution in OpCode#execute
+      //not sure how this would affect other opcodes
+      val availableGas = state.gas - (constGas + varGas(state))
+      val startGas = allBut64th(availableGas)
+
+      val context = ProgramContext[W, S](newEnv, startGas, world2)
+      val result = VM.run(context)
+
+      val codeDepositGas = GasFee.calcCodeDepositCost(result.returnData)
+      val gasUsed = startGas - result.gasRemaining + codeDepositGas
+
+      if (result.error.isDefined) {
+        val stack2 = stack1.push(UInt256.Zero)
+        state.withStack(stack2).spendGas(startGas)
+      } else if (gasUsed > availableGas) {
+        state.withError(OutOfGas)
+      } else {
+        val stack2 = stack1.push(newAddress.toUInt256)
+        val world3 = result.world.saveCode(newAddress, result.returnData)
+        state
+          .withStack(stack2)
+          .withWorld(world3)
+          .withAddressesToDelete(result.addressesToDelete)
+          .withMemory(memory1)
+          .spendGas(gasUsed)
+          .step()
```

Comment:
Should the new state also have the logs that resulted from the execution of `initCode`? According to the YP it seems to be that way.